doc/kj.md

```diff
@@ -82,3 +82,69 @@ struct Empty {};
 
 kj::OneOf<Empty, double, kj::String> foo;
 ```
+
+### Own (unique_ptr)
+
+`Own<T>` is a template for a smart pointer that ensures that only a single owner holds a given
+pointer at a given point in time, similar to
+[`std::unique_ptr`](https://en.cppreference.com/w/cpp/memory/unique_ptr). It has the advantage
+of not allowing manual `release()` of the owned memory and only allowing automatic memory
+management, thus preventing accidental double-frees and facilitating the trivial implementation
+of custom memory allocators if and when needed.
+
+For the general case, you can construct an `Own<T>` using `kj::heap<T>(Args ...)`, which allocates
+memory on the heap and constructs an instance of `T` by calling the constructor matching the
+provided arguments:
+
+```c++
+#include <kj/memory.h>
+
+class Thing {
+  ...
+
+  Thing(unsigned quantity, double difficulty);
+
+  ...
+};
+
+auto ownedThing = kj::heap<Thing>(1, 7.5);
+```
+
+You can also use it as part of a class, in which case it's default initialized to `nullptr` until
+you initialize it otherwise:
+
+```c++
+class Continent {
+  ...
+
+  Continent(Thing creature) : creature(creature) {};
+
+  ...
+
+  kj::Own<Thing> creature;
+
+  ...
+};
+
+Continent antarctica(kj::heap<Thing>(3, 999));
+```
+
+Since an `Own<T>` is fundamentally a wrapper around a pointer, you can use it in the same manner as
+a raw pointer when attempting to access its' internals:
+
+```c++
+if (ownedThing) {
+  ownedThing->morph();
+}
+```
+
+Meanwhile, if you'd like to transfer ownership of the underlying pointer, you can simply move your
+existing `Own<T>` into another, which will set the original (existing) `Own<T>` to `nullptr`:
+
+```c++
+transport(kj::mv(ownedThing))
+```
+
+When the `Own<T>` goes out of scope or when an object containing an `Own<T>` is destructed, the
+destructor for the object pointed to by the `Own<T>` is called and then the underlying memory is
+released automatically.
```

Comment:
That's true most of the time, because usually the object in the `kj::Own` has been allocated with `kj::heap`, which uses a disposer function that calls the destructor.  But there are other ways to create `kj::Own`s that use disposers that don't call the destructor.  For example, `kj::Own`s returned by `kj::refcount()/kj::addRef()` only call the destructor when the reference count drops to zero.  And `kj::Own`s returned by `kj::attachRef()` do nothing when destructed, with the assumption that something else will be taking care of it.  Might be better to just say something like "When an `Own<T>` is destructed, it performs any necessary destruction or deallocation of the object it points to."?